source/common/upstream/upstream_impl.cc

```diff
@@ -1126,6 +1216,18 @@ ClusterInfoImpl::ClusterInfoImpl(
   // If load_balancing_policy is set we will use it directly, ignoring lb_policy.
   if (config.has_load_balancing_policy()) {
     configureLbPolicies(config, server_context);
+  } else if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.convert_legacy_lb_config")) {
```

Comment:
so generally we runtime guard if there's going to be a functional change in behavior.  Is this expected to be a behavioral change, or guarded just because it's high risk?

Reply:
Because the high risk. Ideally, this shouldn't bring any behaviour change.